src/drivers/warmup.ts

```diff
@@ -0,0 +1,129 @@
+   * a warmup request comes in from the same adapter, if it ever is allowed to expire, it means we should send
+   * a warm up response to the external adapter
+   */
+  ttl: number
+  /**
+   * Metadata related to the external adapter itself that we can index, ex. the current response data of the EA that it
+   * has received while warming up its cache
+   */
+  meta: any
+}
+
+/**
+ * These set of events are emitted when a v1 EA calls this service
+ */
+export const warmupRequestRecieved = createAction<WarmupRequestRecievedPayload>(
+  'WARMUP_REQUEST_RECIEVED',
+)
+export const warmupRequestFulfilled = createAction('WARMUP_REQUEST_FULFILLED')
+export const warmupRequestFailed = createAction('WARMUP_REQUEST_FAILED')
+
+interface WarmupResponseRequestedPayload {
+  /**
+   * State lookup key so that the warmup response handler can find the slice of data it needs
+   * to warmup the cold EA
+   */
+  key: string
+}
+interface WarmupResponseFulfilledPayload {
+  /**
+   * State lookup key
+   */
+  key: string
+}
+interface WarmupResponseFailedPayload {
+  /**
+   * State lookup key
+   */
+  key: string
+}
+/**
+ * These set of events are emitted when our warmup driver calls a v1 EA to warm up its cache
+ */
+export const warmupResponseRequested = createAction<WarmupResponseRequestedPayload>(
+  'WARMUP_RESPONSE_REQUESTED',
+)
+export const warmupResponseFulfilled = createAction<WarmupResponseFulfilledPayload>(
+  'WARMUP_RESPONSE_FULFILLED',
+)
+export const warmupResponseFailed = createAction<WarmupResponseFailedPayload>(
+  'WARMUP_RESPONSE_FAILED',
+)
+
+/**
+ * Handle warmup request received events
+ */
+export const warmupRequestEpic: Epic = (action$, _state$) =>
+  action$.pipe(
+    filter(warmupRequestRecieved.match),
+    mergeMap((action) => {
+      const dropWarmupResponse$ = action$.pipe(
+        filter(warmupRequestRecieved.match),
+        filter((a) => isSameEA(a, action)),
+      )
+      // if this emits, we will emit an event to send a warmup response to an EA
+      const requestWarmupResponse$ = of(warmupResponseRequested({ key: hashEA(action) })).pipe(
+        delay(action.payload.ttl),
+      )
+
+      return race(requestWarmupResponse$, dropWarmupResponse$)
+    }),
+    catchError(() => of(warmupRequestFailed())),
+  )
+
+/**
+ * Handle warmup response request events
+ */
+export const warmupResponseEpic: Epic = (action$, state$) =>
+  action$.pipe(
+    filter(warmupResponseRequested.match),
+    withLatestFrom(state$),
+    map(([action, state]) => ({
+      ...(state[action.payload.key] as WarmupRequestRecievedPayload),
+      key: action.payload.key,
+    })),
+    mergeMap(({ url, body, key }) =>
+      axios.post(url, body).pipe(
+        mapTo(key),
+        catchError(() => throwError(key)),
+      ),
+    ), // no timeout for now but we should really add one
+    map((key) => warmupResponseFulfilled({ key })),
```

Comment:
This is a transaction into the system. An event that happened that is important to store in the event store -> new data is observed. This event should carry all change data, not just a key to lookup in the state. We need all info in the event so we can recreate that state from the event log if we need/want to reply.

Reply:
do we need a "fat" event here? The original warmup cmd should store all the info we need at that key